### PR TITLE
[paperles] Set PAPERLES_URL automatically

### LINKED ISSUE #1538: [paperless] CSRF verification failed
### Helm chart name

paperless

### Helm chart version

8.6.2

### Container name

ghcr.io/paperless-ngx/paperless-ngx:ngx-1.7.0

### Container tag

ngx-1.7.0

### Description

When installing Paperless with Ingress enabled, I get `Your CSRF verification failed` when I try to browse to the app

### Expected result

I expect to be able to access the app via the Ingress

### Helm values to reproduce

```YAML
ingress:
  main:
    enabled: true
    annotations:
      cert-manager.io/cluster-issuer: "letsencrypt-prod"
      nginx.ingress.kubernetes.io/proxy-body-size: "0"
      nginx.ingress.kubernetes.io/proxy-buffering: "off"
    ingressClassName: "public"
    hosts:
      - host: paperless.example.com
        paths:
          - path: /
            pathType: Prefix
    tls:
      - secretName: paperless-tls
        hosts:
          - paperless.example.com
```

### Additional Information

This seems to be discussed upstream in https://github.com/paperless-ngx/paperless-ngx/issues/712

The solution is to set the environment variable `PAPERLESS_URL` to the same value as the Ingress host.

In my case, setting

```yaml
env:
  PAPERLESS_URL: https://paperless.example.com
```

made the app work via the Ingress as expected, so I think there should be at least a placeholder for `PAPERLESS_URL` in `values.yaml`. Even better would be the ability to automatically set `PAPERLESS_URL` to the same value as `ingress.main.hosts.host[0]`

### Repo link

_No response_

### PR DESCRIPTION
**Pull request by angelnu**
**Description of the change**

Set PAPERLESS_URL by default to the main ingress host - see https://github.com/paperless-ngx/paperless-ngx/issues/712

**Benefits**

Working helm chart after upgrading to paperless-ngx >=v1.7

**Possible drawbacks**

None

**Applicable issues**

- fixes #1538 

**Additional information**

**Checklist** 
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [x] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

File: charts/stable/paperless/values.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ env:
   # PAPERLESS_ADMIN_USER: admin
   # -- Password for the root user
   # PAPERLESS_ADMIN_PASSWORD: admin
+  # PAPERLESS_URL: <set do main ingress by default>
```
Comment (bjw-s) [inline charts/stable/paperless/values.yaml:39]:
```suggestion
  # PAPERLESS_URL: <set to main ingress by default>
```

File: charts/stable/paperless/templates/common.yaml [hunk 2]
```diff
@@ -1 +1,11 @@
+{{/* Make sure all variables are set properly */}}
+{{- include "common.values.setup" . }}
+
+{{/* Append the hardcoded settings */}}
+{{- define "paperless.harcodedValues" -}}
+env:
+    PAPERLESS_URL: {{ (first .Values.ingress.main.hosts).host }}
```
Comment (bjw-s) [inline charts/stable/paperless/templates/common.yaml:7]: Wouldn't doing it like this always override any custom values that have been set by the user?

**Review by bjw-s — CHANGES_REQUESTED** [review]

File: charts/stable/paperless/templates/common.yaml (hunk 2 above)
Comment (angelnu) [reply charts/stable/paperless/templates/common.yaml:7]: I did a merge instead of mergeOverride to avoid overriding user values

Comment (angelnu) [reply charts/stable/paperless/templates/common.yaml:7]: .. but override is not doing what I was expecting...

**Review by bjw-s — APPROVED** [review]

**Pull request merged**